sale_pricelist_auto_update/views/res_partner_views.xml

```diff
@@ -20,6 +20,10 @@
                     </field>
                 </page>
             </xpath>
+            <xpath expr="//group[@name='sale']" position="inside">
+                <field name="fix_pricelist"
+                       attrs="{'invisible': [('is_company','=',False),('parent_id','!=',False)]}"/>
```

Comment:
You can extend the idea further to do the same for the yearly purchase history tab.  No need to show it for non-commercial partners.